backend/src/v2/driver/driver.go

```diff
@@ -749,6 +756,33 @@ func DAG(ctx context.Context, opts Options, mlmd *metadata.Client) (execution *E
 	ecfg.ParentDagID = dag.Execution.GetID()
 	ecfg.IterationIndex = iterationIndex
 	ecfg.NotTriggered = !execution.WillTrigger()
+
+	// Handle writing output parameters to MLMD.
+	outputParameters := opts.Component.GetDag().GetOutputs().GetParameters()
+	glog.V(4).Info("outputParameters: ", outputParameters)
+	for _, value := range outputParameters {
+		outputParameterKey := value.GetValueFromParameter().OutputParameterKey
+		producerSubTask := value.GetValueFromParameter().ProducerSubtask
+		glog.V(4).Info("outputParameterKey: ", outputParameterKey)
+		glog.V(4).Info("producerSubtask: ", producerSubTask)
+
+		outputParameterMap := map[string]interface{}{
+			"output_parameter_key": outputParameterKey,
+			"producer_subtask":     producerSubTask,
+		}
+
+		outputParameterStruct, _ := structpb.NewValue(outputParameterMap)
+
+		ecfg.OutputParameters = map[string]*structpb.Value{
+			value.GetValueFromParameter().OutputParameterKey: outputParameterStruct,
+		}
```

Comment:
this looks like you are re writing ecfg.OutputParameters with a new map every iteration, did you mean to update the map instead?

Reply:
You're absolutely right! We would likely have hit a wall because of this when we started tested multiple outputs.

Really appreciate the time and effort you took to grok some not particularly grokkable code. Thank you for the feedback, @HumairAK!

Reply:
Anytime! feel free to @ me here or slack if you'd like another review

Reply:
Awesome! I spent like 2 hours trying to dig up where the overwriting of outputs was happening! Thank @HumairAK